core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -319,6 +318,105 @@ object Uri extends UriFunctions {
       }
   }
 
+  sealed abstract class Host {
+    def inetAddress: Either[UnknownHostException, InetAddress]
+    def value: String
+    def render(w: Writer): w.type
+
+    final override def equals(o: Any) = o match {
+      case that: Host => this.value.equalsIgnoreCase(that.value)
```

Comment:
This should probably compare the encoded values, or else one could do stupid tricks with percent encoding.  Alternatively, we could reintroduce private subclasses.

Reply:
But then we'd need to normalize, too.  Bah.